common-kafka/src/main/java/com/cerner/common/kafka/consumer/ProcessingKafkaConsumer.java

```diff
@@ -194,6 +195,21 @@ public ProcessingKafkaConsumer(ProcessingConfig config) {
         this(config, new KafkaConsumer<>(config.getProperties()));
     }
 
+    /**
+     * Creates a new {@link ProcessingKafkaConsumer} with the given configuration and deserializers using the {@link KafkaConsumer}.
+     *
+     * @param config the configuration used by the consumer
+     * @param keyDeserializer The deserializer instance for value. When passing the deserializer instance
+     *          directly, the {@link Deserializer#configure(Map, boolean)} method will not be called.
+     * @param valueDeserializer The deserializer instance for value. When passing the deserializer instance
+     *          directly, the {@link Deserializer#configure(Map, boolean)} method will not be called.
+     * @throws NullPointerException if config is {@code null}
```

Comment:
I documented this as `NullPointerException` because the precondition that would throw an `IllegalArgumentException` is never reached due to the call to `config.getProperties`. The existing public constructor would also through a `NullPointerException` but is currently documented as throwing an `IllegalArgumentException`.

Reply:
would you mind fixing that in both cases?

a new static method extracting the properties from the config could throw the IAE